src/Security/Security.php

```diff
@@ -269,7 +269,7 @@ public function index()
      */
     protected function getAuthenticator($name = 'default')
     {
-        $authenticators = $this->authenticators;
+        $authenticators = Security::singleton()->getAuthenticators();
```

Comment:
getAuthenticators() is already an instance method on the service with the assigned dependencies. Don't re-invoke singleton() here. I can understand it on CMSSecurity though, but not in Security, please.